### ISSUE TITLE
BUGFIX: for image batch (more than 1 image) with attention masks

### ISSUE THREAD

**Issue by tolly-andreas** [lldacing/ComfyUI_PuLID_Flux_ll#53]
Great job on providing PuLID in ComfyUI! Thanks a lot!

When experimenting, I've encountered an error when trying to compute a batch of multiple latent images and using attn_mask inputs.

Currently in line 44 of PulidFluxHook.py a 1 is hardcoded as batch size for the mask:

mask = mask[..., 0].unsqueeze(-1).repeat(1, 1, pulid_img.shape[-1]).to(dtype=pulid_img.dtype)
            
this should be:
mask = mask[..., 0].unsqueeze(-1).repeat(**bs**, 1, pulid_img.shape[-1]).to(dtype=pulid_img.dtype)

Do you want me to create a PR for this? Its really just the one line. 
Keep up the great work! 
Cheers, Andreas

-------------------
Exception trace for trying to run a batch size of 3:

"exception_message": "The size of tensor a (6) must match the size of tensor b (3) at non-singleton dimension 0",
    "exception_type": "RuntimeError",
    "traceback": [
      "  File \"[/src/ComfyUI/execution.py](http://localhost:8888/src/ComfyUI/execution.py)\", line 327, in execute\n    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)\n                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/execution.py](http://localhost:8888/src/ComfyUI/execution.py)\", line 202, in get_output_data\n    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)\n                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/execution.py](http://localhost:8888/src/ComfyUI/execution.py)\", line 174, in _map_node_over_list\n    process_inputs(input_dict, i)\n",
      "  File \"[/src/ComfyUI/execution.py](http://localhost:8888/src/ComfyUI/execution.py)\", line 163, in process_inputs\n    results.append(getattr(obj, func)(**inputs))\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy_extras/nodes_custom_sampler.py](http://localhost:8888/src/ComfyUI/comfy_extras/nodes_custom_sampler.py)\", line 657, in sample\n    samples = guider.sample(noise.generate_noise(latent), latent_image, sampler, sigmas, denoise_mask=noise_mask, callback=callback, disable_pbar=disable_pbar, seed=noise.seed)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 1008, in sample\n    output = executor.execute(noise, latent_image, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 111, in execute\n    return self.wrappers[self.idx](self, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/pulidflux.py](http://localhost:8888/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/pulidflux.py)\", line 613, in pulid_outer_sample_wrappers_with_override\n    out = wrapper_executor(noise, latent_image, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 103, in __call__\n    return new_executor.execute(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 110, in execute\n    return self.original(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 976, in outer_sample\n    output = self.inner_sample(noise, latent_image, device, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 959, in inner_sample\n    samples = executor.execute(self, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 110, in execute\n    return self.original(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 738, in sample\n    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/root/.pyenv/versions/3.12.6/lib/python3.12/site-packages/torch/utils/_contextlib.py](http://localhost:8888/root/.pyenv/versions/3.12.6/lib/python3.12/site-packages/torch/utils/_contextlib.py)\", line 116, in decorate_context\n    return func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/k_diffusion/sampling.py](http://localhost:8888/src/ComfyUI/comfy/k_diffusion/sampling.py)\", line 161, in sample_euler\n    denoised = model(x, sigma_hat * s_in, **extra_args)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 390, in __call__\n    out = self.inner_model(x, sigma, model_options=model_options, seed=seed)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 939, in __call__\n    return self.predict_noise(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 942, in predict_noise\n    return sampling_function(self.inner_model, x, timestep, self.conds.get(\"negative\", None), self.conds.get(\"positive\", None), self.cfg, model_options=model_options, seed=seed)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 370, in sampling_function\n    out = calc_cond_batch(model, conds, x, timestep, model_options)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 206, in calc_cond_batch\n    return executor.execute(model, conds, x_in, timestep, model_options)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 110, in execute\n    return self.original(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/samplers.py](http://localhost:8888/src/ComfyUI/comfy/samplers.py)\", line 319, in _calc_cond_batch\n    output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/model_base.py](http://localhost:8888/src/ComfyUI/comfy/model_base.py)\", line 133, in apply_model\n    return comfy.patcher_extension.WrapperExecutor.new_class_executor(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 111, in execute\n    return self.wrappers[self.idx](self, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/pulidflux.py](http://localhost:8888/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/pulidflux.py)\", line 638, in pulid_apply_model_wrappers\n    out = wrapper_executor(x, t, c_concat, c_crossattn, control, transformer_options, **kwargs)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 103, in __call__\n    return new_executor.execute(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/patcher_extension.py](http://localhost:8888/src/ComfyUI/comfy/patcher_extension.py)\", line 110, in execute\n    return self.original(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/model_base.py](http://localhost:8888/src/ComfyUI/comfy/model_base.py)\", line 165, in _apply_model\n    model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds).float()\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/root/.pyenv/versions/3.12.6/lib/python3.12/site-packages/torch/nn/modules/module.py](http://localhost:8888/root/.pyenv/versions/3.12.6/lib/python3.12/site-packages/torch/nn/modules/module.py)\", line 1739, in _wrapped_call_impl\n    return self._call_impl(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/root/.pyenv/versions/3.12.6/lib/python3.12/site-packages/torch/nn/modules/module.py](http://localhost:8888/root/.pyenv/versions/3.12.6/lib/python3.12/site-packages/torch/nn/modules/module.py)\", line 1750, in _call_impl\n    return forward_call(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/comfy/ldm/flux/model.py](http://localhost:8888/src/ComfyUI/comfy/ldm/flux/model.py)\", line 203, in forward\n    out = self.forward_orig(img, img_ids, context, txt_ids, timestep, y, guidance, control, transformer_options, attn_mask=kwargs.get(\"attention_mask\", None))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/PulidFluxHook.py](http://localhost:8888/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/PulidFluxHook.py)\", line 185, in pulid_forward_orig\n    out = blocks_replace[(\"double_block\", i)]({\"img\": img,\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/PulidFluxHook.py](http://localhost:8888/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/PulidFluxHook.py)\", line 74, in __call__\n    img = img + callback(temp_img,\n                ^^^^^^^^^^^^^^^^^^\n",
      "  File \"[/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/PulidFluxHook.py](http://localhost:8888/src/ComfyUI/custom_nodes/ComfyUI_PuLID_Flux_ll/PulidFluxHook.py)\", line 47, in pulid_patch\n    pulid_img = pulid_img * mask\n                ~~~~~~~~~~^~~~~~\n"
    ],

**Comment by tolly-andreas** [comment]
Actually, I could get around this issue in my workflow, by following the example and using the 
FluxForwardOverrider and TeaCache 
from the patches. Also saw that my "bugfix" would break the other examples.
